core/lib/dal/src/storage_logs_dal.rs

```diff
@@ -68,6 +69,46 @@ impl StorageLogsDal<'_, '_> {
         copy.finish().await.unwrap();
     }
 
+    pub async fn insert_storage_logs_from_snapshot(
+        &mut self,
+        miniblock_number: MiniblockNumber,
+        snapshot_storage_logs: &[SnapshotStorageLog],
+    ) {
+        let mut copy = self
+            .storage
+            .conn()
+            .copy_in_raw(
+                "COPY storage_logs(
+                    hashed_key, address, key, value, operation_number, tx_hash, miniblock_number,
+                    created_at, updated_at
+                )
+                FROM STDIN WITH (DELIMITER '|')",
+            )
+            .await
+            .unwrap();
+
+        let mut buffer = String::new();
+        let now = Utc::now().naive_utc().to_string();
+        for log in snapshot_storage_logs.iter() {
+            write_str!(
+                &mut buffer,
+                r"\\x{hashed_key:x}|\\x{address:x}|\\x{key:x}|\\x{value:x}|",
+                hashed_key = log.key.hashed_key(),
+                address = log.key.address(),
+                key = log.key.key(),
+                value = log.value
+            );
+            writeln_str!(
+                &mut buffer,
+                r"{}|\\x{:x}|{miniblock_number}|{now}|{now}",
+                log.enumeration_index,
+                H256::zero()
```

Comment:
Ah yes, there's a FIXME in the snapshot syncing spec regarding `tx_hash`. I'm actually not entirely sure that fake `tx_hash`es will work as expected; there are at least some DB queries that rely on `tx_hash` values being correct (one of which, `get_transaction_receipt()`, you've touched in this PR).